recastjs/src/recastjs.cpp

```diff
@@ -71,6 +91,282 @@ void NavMesh::destroy()
     }
     dtFreeNavMesh(m_navMesh);
     dtFreeNavMeshQuery(m_navQuery);
+    if (m_tileCache)
+    {
+        dtFreeTileCache(m_tileCache);
+    }
+    m_talloc.reset();
+}
+
+int NavMesh::rasterizeTileLayers(const int tx, const int ty,
+                                    const rcConfig& cfg,
+                                    TileCacheData* tiles,
+                                    const int maxTiles,
+                                    NavMeshintermediates& intermediates,
+                                    const std::vector<unsigned char>& triareas,
+                                    const std::vector<float>& verts)
+{
+    RecastFastLZCompressor comp;
+
+    // Tile bounds.
+    const float tcs = cfg.tileSize * cfg.cs;
+    
+    rcConfig tcfg;
+    memcpy(&tcfg, &cfg, sizeof(tcfg));
+
+    tcfg.bmin[0] = cfg.bmin[0] + tx*tcs;
+    tcfg.bmin[1] = cfg.bmin[1];
+    tcfg.bmin[2] = cfg.bmin[2] + ty*tcs;
+    tcfg.bmax[0] = cfg.bmin[0] + (tx+1)*tcs;
+    tcfg.bmax[1] = cfg.bmax[1];
+    tcfg.bmax[2] = cfg.bmin[2] + (ty+1)*tcs;
+    tcfg.bmin[0] -= tcfg.borderSize*tcfg.cs;
+    tcfg.bmin[2] -= tcfg.borderSize*tcfg.cs;
+    tcfg.bmax[0] += tcfg.borderSize*tcfg.cs;
+    tcfg.bmax[2] += tcfg.borderSize*tcfg.cs;
+    
+    NavMeshintermediates tileIntermediates;
+    // Allocate voxel heightfield where we rasterize our input data to.
+    tileIntermediates.m_solid = rcAllocHeightfield();
+    if (!tileIntermediates.m_solid)
+    {
+        Log("buildNavigation: Out of memory 'solid'.");
+        return 0;
+    }
+    rcContext ctx;
+
+    if (!rcCreateHeightfield(&ctx, *tileIntermediates.m_solid, tcfg.width, tcfg.height, tcfg.bmin, tcfg.bmax, tcfg.cs, tcfg.ch))
+    {
+        Log("buildNavigation: Could not create solid heightfield.");
+        return 0;
+    }
+    
+    rcChunkyTriMesh* chunkyMesh = intermediates.m_chunkyMesh;
+
+    float tbmin[2], tbmax[2];
+    tbmin[0] = tcfg.bmin[0];
+    tbmin[1] = tcfg.bmin[2];
+    tbmax[0] = tcfg.bmax[0];
+    tbmax[1] = tcfg.bmax[2];
+    int cid[512];// TODO: Make grow when returning too many items.
+    const int ncid = rcGetChunksOverlappingRect(chunkyMesh, tbmin, tbmax, cid, 512);
```

Comment:
Waiting for the TODO, maybe log a message when the number of returned items is 512 for tracking/debugging purpose?

Reply:
won't fix. The use case to get more than 512 chunks is obscure. I will fix it when I encounter an issue (if ever)